src/main/java/frc/robot/lib/RoboLionsPID.java

```diff
@@ -100,60 +97,39 @@ public double execute(double command, double feedback) {
     // 3. Latch internal error's state for next time
     previousError = error; 
 
-    // 4. If in deadband, don't integrate the error
+    // 4. If in deadband, PID output = 0
     if (enableDeadBand == true) {
-      if((error < upperDeadbandLimit) && (error > lowerDeadbandLimit)) {
-        deadband_counter++;
-        if(deadband_counter > 13) { // 13 = 50 / 4 = 50 ms split into 4 quarters
-          deadband_active = true;
-          deadband_counter = 13;
-        }
-          // integral_charge = integral_charge + error*deltaTime;
+      if ((error <= upperDeadbandLimit) && (error >= lowerDeadbandLimit)) {
+        return output;
       }
     }
     
-    // 5. If in Zone of Contrability, overcome any frictional forces
+    // 5. If in cage limit, prevent integral from increasing further
+    integral_charge += error*deltaTime;
     if (enableCage == true) {
-      if((error < upperCageLimit) && (error > lowerCageLimit)){
-        //integrate if error is small and we are close to our command
-        deadband_counter = 0;
-        integral_charge = integral_charge + error*deltaTime;
-        deadband_active = false; //true
-      }
+      Math.min(integral_charge, upperCageLimit);
+      Math.max(integral_charge, lowerCageLimit);
     }
 
-    else {
-      integral_charge = integral_charge + error*deltaTime;
-    }
-
-    // 6. If in Zone of Saturation, you can't move any faster, so scale off the throttle 
-    /* else {
-       integral_charge = 0;
-       deadband_counter = 0;
-       deadband_active = false; //true
-       //integrate if error is small and we are close to our command
-    }
-    */
-
-    // 7. Calculate PID Control Equation
+    // 6. Calculate PID Control Equation
     output = error*proportionalGain + integral_charge*integralGain - derivativeCalculation*derivativeGain;
-    if(output > maxOutput) {
+    if (output > maxOutput) {
```

Comment:
Just like you did on line 110, you can simplify this if else case by utilizing Math.min / Math.max.